warp10/src/main/java/io/warp10/continuum/gts/MetadataSelectorMatcher.java

```diff
@@ -177,8 +212,19 @@ public boolean matches(Metadata metadata) {
     } else {
       // standard selector
       if (null != this.labelsPatterns) {
+        for (String label: this.missingLabels) {
+          if (null != inputLabels && null != inputLabels.get(label)) {
+            labelAndAttributeMatch = false;
+            break;
+          } else if (null != inputAttributes && null != inputLabels.get(label)) {
```

Comment:
Shouldn't  it be `null != inputAttributes && null != inputAttributes.get(label)` ?
Also you could merge the if and the else if blocks because they do the same.